### ISSUE TITLE
Allow templating skipCrds key

### ISSUE THREAD

**Issue by diversario** [argoproj/argo-cd#17146]
# Summary

Given a source like this:

```
spec:
  sources:
  - repoURL: "https://prometheus-community.github.io/helm-charts"
    chart: kube-prometheus-stack
    targetRevision: 45.31.1
    helm:
      skipCrds: true
```

I would like to be able to put a go template as the value of the `skipCrds` key:

```
spec:
  sources:
  - repoURL: "https://prometheus-community.github.io/helm-charts"
    chart: kube-prometheus-stack
    targetRevision: 45.31.1
    helm:
      skipCrds: {{ .someValueFromGenerators }}
```

# Motivation

In my case, the use case is moving away from installing CRDs separately from a helm chart. The example is real – currently, I'm installing prometheus operator CRDs elsewhere and they are disabled in the app, but I would like to start using the CRDs in installations in new clusters. Templating would allow me to do something like

```
      skipCrds: {{ .k8sClusterVersion > 1.25 }}
```

At the moment, I have no choice but to copy the appset under a different name, which causes issues in other places (for example, different release name means Service names are different, and there are other apps that expect certain service names).

# Proposal

I'm not sure about the implementation, but hopefully it's quite similar to how existing fields are templated.

**Comment by daftping** [comment]
Only strings can be templated. `skipCrds` is boolean.

You can use `templatePatch` released in 2.10.0
https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Template/#template-patch
```yaml
  templatePatch: |
    spec:
      sources:
      - repoURL: "https://prometheus-community.github.io/helm-charts"
        chart: kube-prometheus-stack
        targetRevision: 45.31.1
        helm:
          skipCrds: {{ .someValueFromGenerators }}
```
Please note, that you have to put the whole `sources` key into `templatePatch` due to how lists are merged.
https://github.com/argoproj/argo-cd/issues/16800

**Comment by diversario** [comment]
Ah, nice, that should work fine for our use case. Will try out once we've upgraded. Thanks!
